src/librustc_ast_lowering/lib.rs

```diff
@@ -170,7 +169,9 @@ struct LoweringContext<'a, 'hir: 'a> {
 }
 
 pub trait Resolver {
-    fn cstore(&self) -> &dyn CrateStore;
+    fn def_key(&mut self, id: DefId) -> DefKey;
+
+    fn item_generics_cloned_untracked_liftimes(&self, def: DefId, sess: &Session) -> usize;
```

Comment:
It's possible that `fn item_generics_cloned_untracked` and `fn def_key` can be removed from `trait CrateStore` now.

Reply:
`item_generics_cloned_untracked` certainly can.

Reply:
Not sure I follow what you mean here. The implementation in `impl rustc_ast_lowering::Resolver for Resolver<'_> {` still needs to access `item_generics_cloned_untracked` from `self.cstore()`?

Reply:
Now it can be an inherent method on `CStore` because `rustc_resolve` can access `CStore` directly without anonymizing it with `trait CrateStore`.

Reply:
Ah, great, thanks! I added `item_generics_num_liftimes` directly to `CStore` to expose as little info as possible.